client/components/external-link/index.jsx

```diff
@@ -36,7 +36,7 @@ export default React.createClass( {
 			'has-icon': !! this.props.icon,
 		} );
 
-		const props = assign( {}, this.props, {
+		const props = assign( {}, omit( this.props, 'icon', 'iconSize' ), {
```

Comment:
Wouldn't it be more future-proof if we pick props we want/need instead of omitting these two? If somebody passes something not valid for `<a>` and we don't omit it here, we'll get the error again.